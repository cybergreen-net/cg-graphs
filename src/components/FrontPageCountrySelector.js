import React, { Component } from 'react';
import Select from 'react-select';
import 'react-select/dist/react-select.css';

export default class FrontPageCountrySelector extends Component {
  constructor(props) {
    super(props)
    this.state = {
      selectedCountry: undefined
    }
  }

  onChange(country) {
    // purely for testing atm - we don't use state internally
    this.setState({
      selectedCountry: country
    });
    if (country.slug) {
      window.location = `/country/${country.slug}`
    }
  }

  render() {
    const selectOptions = Object.values(this.props.countries).map(country => {
      return {
        value: country.id,
        label: country.name,
        slug: country.slug
      }
    })

    selectOptions.unshift({value: '', label: 'Select a country'})
    return (
      <div>
        <Select
          name="country-selector"
          value={this.state.selectedCountry || selectOptions[0]}
          options={selectOptions}
          onChange={this.onChange.bind(this)}
        />
      </div>
    );
  }
}
