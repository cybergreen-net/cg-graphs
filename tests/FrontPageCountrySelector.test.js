import FrontPageCountrySelector from '../src/components/FrontPageCountrySelector';
import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('FrontPageCountrySelector component is working fine', () => {
  function setup() {
    let props = {
      countries: {
        't': {id: 't', name: 'Global', slug: ''},
        'ge': {id: 'ge', name: 'Georgia', slug: 'georgia'},
        'kz': {id: 'kz', name: 'Kazakhstan', slug: 'Kazakhstan'},
        'gb': {id: 'gb', name: 'United Kingdom', slug: 'united-kingdom'},
        'us': {id: 'us', name: 'United States', slug: 'united-states'}
      }
    }
    const enzymeWrapper = shallow(< FrontPageCountrySelector {...props} />)

    return {
      props,
      enzymeWrapper
    }
  }

  it('Renders expected DOM', () => {
    const { enzymeWrapper } = setup()
    expect(toJson(enzymeWrapper)).toMatchSnapshot();
  })

  it('When a country is selected, it updates state of the component', () => {
    const { enzymeWrapper, props } = setup()
    let newCountry = {value: 'us', label: 'United States', slug: 'united-states'}
    expect(enzymeWrapper.state().selectedCountry).toEqual(undefined)
    enzymeWrapper.find('Select').simulate('change', newCountry)
    expect(enzymeWrapper.state().selectedCountry).toEqual(newCountry)
  })

})
