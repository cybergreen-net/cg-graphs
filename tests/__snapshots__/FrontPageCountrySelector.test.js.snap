exports[`FrontPageCountrySelector component is working fine Renders expected DOM 1`] = `
<div>
  <Select
    addLabelText="Add \"{label}\"?"
    arrowRenderer={[Function]}
    autosize={true}
    backspaceRemoves={true}
    backspaceToRemoveMessage="Press backspace to remove {label}"
    clearAllText="Clear all"
    clearRenderer={[Function]}
    clearValueText="Clear value"
    clearable={true}
    deleteRemoves={true}
    delimiter=","
    disabled={false}
    escapeClearsValue={true}
    filterOptions={[Function]}
    ignoreAccents={true}
    ignoreCase={true}
    inputProps={Object {}}
    isLoading={false}
    joinValues={false}
    labelKey="label"
    matchPos="any"
    matchProp="any"
    menuBuffer={0}
    menuRenderer={[Function]}
    multi={false}
    name="country-selector"
    noResultsText="No results found"
    onBlurResetsInput={true}
    onChange={[Function]}
    onCloseResetsInput={true}
    openAfterFocus={false}
    optionComponent={[Function]}
    options={
      Array [
        Object {
          "label": "Select a country",
          "value": "",
        },
        Object {
          "label": "Global",
          "slug": "",
          "value": "t",
        },
        Object {
          "label": "Georgia",
          "slug": "georgia",
          "value": "ge",
        },
        Object {
          "label": "Kazakhstan",
          "slug": "Kazakhstan",
          "value": "kz",
        },
        Object {
          "label": "United Kingdom",
          "slug": "united-kingdom",
          "value": "gb",
        },
        Object {
          "label": "United States",
          "slug": "united-states",
          "value": "us",
        },
      ]
    }
    pageSize={5}
    placeholder="Select..."
    required={false}
    scrollMenuIntoView={true}
    searchable={true}
    simpleValue={false}
    tabSelectsValue={true}
    value={
      Object {
        "label": "Select a country",
        "value": "",
      }
    }
    valueComponent={[Function]}
    valueKey="value" />
</div>
`;
